core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -835,6 +844,7 @@ public void run()
             failure = closeAndAddSuppressed(failure, nodeAllocator);
 
             failure.ifPresent(queryStateMachine::transitionToFailed);
+            schedulerSpan.end();
```

Comment:
should we attach `failure` to `schedulerSpan` if present?

Reply:
I added a new event that will write the exception's getMessage() to a new FAILURE attribute key